### PR TITLE
backport: Merge bitcoin#23642, 22794, 23316, 24365, gui#517, 24219, 23253, 24449, 22543

### PR DESCRIPTION
**Pull request by vijaydasmp**
Bitcoin Backports

### CODE REVIEW

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -869,29 +869,33 @@ static RPCHelpMan decoderawtransaction()
...
+        "\nDecode a hex-encoded script.\n",
+        {
+            {"hexstring", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "the hex-encoded script"},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                {RPCResult::Type::STR, "asm", "Script public key"},
+                {RPCResult::Type::STR, "type", "The output type (e.g. " + GetAllOutputTypes() + ")"},
+                {RPCResult::Type::STR, "address", /* optional */ true, "The Bitcoin address (only if a well-defined address exists)"},
```
Comment (knst) [inline src/rpc/rawtransaction.cpp:883]: 23642: missing dashification in this line and below: Bitcoin -> Dash

File: src/rpc/rawtransaction.cpp [hunk 2]
```diff
@@ -869,29 +869,33 @@ static RPCHelpMan decoderawtransaction()
...
+                {RPCResult::Type::STR, "p2sh", /* optional */ true, "address of P2SH script wrapping this redeem script (not returned if the script is already a P2SH)"},
+                {RPCResult::Type::OBJ, "segwit", /* optional */ true, "Result of a witness script public key wrapping this redeem script (not returned if the script is a P2SH or witness)",
+                 {
+                     {RPCResult::Type::STR, "asm", "String representation of the script public key"},
+                     {RPCResult::Type::STR_HEX, "hex", "Hex string of the script public key"},
+                     {RPCResult::Type::STR, "type", "The type of the script public key (e.g. witness_v0_keyhash or witness_v0_scripthash)"},
```
Comment (knst) [inline src/rpc/rawtransaction.cpp:889]: 23642: missing dashification we don't have type here

File: src/rpc/rawtransaction.cpp [hunk 3]
```diff
@@ -869,29 +869,33 @@ static RPCHelpMan decoderawtransaction()
...
+                     {RPCResult::Type::STR, "address", /* optional */ true, "The Bitcoin address (only if a well-defined address exists)"},
+                     {RPCResult::Type::STR, "p2sh-segwit", "address of the P2SH script wrapping this witness redeem script"},
```
Comment (knst) [inline src/rpc/rawtransaction.cpp:891]: 23642: missing dashification no segwit for dash

**Review by knst — CHANGES_REQUESTED** [review]

File: src/rpc/rawtransaction.cpp [hunk 4]
```diff
@@ -869,29 +869,31 @@ static RPCHelpMan decoderawtransaction()
...
+                {RPCResult::Type::STR, "type", "The output type (e.g. " + GetAllOutputTypes() + ")"},
+                {RPCResult::Type::STR, "address", /* optional */ true, "The Dash address (only if a well-defined address exists)"},
+                {RPCResult::Type::STR, "p2sh", /* optional */ true, "address of P2SH script wrapping this redeem script (not returned if the script is already a P2SH)"},
+                {RPCResult::Type::OBJ, "segwit", /* optional */ true, "Result of a witness script public key wrapping this redeem script (not returned if the script is a P2SH or witness)",
+                 {
+                     {RPCResult::Type::STR, "asm", "String representation of the script public key"},
+                     {RPCResult::Type::STR_HEX, "hex", "Hex string of the script public key"},
+                     {RPCResult::Type::STR, "address", /* optional */ true, "The Dash address (only if a well-defined address exists)"},
+                 }},
+            },
```
Comment (knst) [inline src/rpc/rawtransaction.cpp:891]: all of these lines should not have any diff except whitespaces, but it's removed substrings "DEPRECATED", added "segwit", etc... Please check carefully

original patch (ignoring whitespaces chages):
image
vs your patch (ignoring whitespaces changes):
image

Comment (knst) [reply src/rpc/rawtransaction.cpp:891]: @vijaydasmp you can use feature `git diff -b` and `git show -b` to compare diff without whitespace-changes or use similar feature on GitHub to do self-review of changes
image

**Review by knst — APPROVED** [review]
LGTM 0cf0b9135afc2eba700469c3ea5ccf6e2e660473

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta @UdjinM6 requesting review

**Review by UdjinM6 — APPROVED** [review]
utACK 39316088585984faf4ea422745e64f0adef916f8

**Comment by vijaydasmp** [comment]
Hello @knst @PastaPastaPasta , requesting review

**Comment by vijaydasmp** [comment]
Hello @knst @PastaPastaPasta , requesting review

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta requesting review

**Review by PastaPastaPasta — APPROVED** [review]
utACK 39316088585984faf4ea422745e64f0adef916f8

**Pull request merged**